build/sage_bootstrap/download/mirror_list.py

```diff
@@ -190,14 +188,18 @@ def _rank_mirrors(self):
             result_ms = int(1000 * result)
             log.info(str(result_ms).rjust(5) + 'ms: ' + mirror)
             timed_mirrors.append((result, mirror))
+            timed_mirrors.sort()
+            if len(timed_mirrors) >= 5 and timed_mirrors[4][0] < 0.3:
+                # We don't need more than 5 decent mirrors
+                break
+
```

Comment:
Where that number 5 comes from?

Reply:
Just feeling confident that when 5 mirrors say they don't have the file, it's time to contact the upstream url.

Reply:
As I don't know how things here work, I guessed that it is related with the maximum concurrent number of downloads of packages and wondered how that number is determined to be 5...

Reply:
OK. Now I learned a little how things work here. A "decent" mirror is a site that allows fast connection and hence the number of them has nothing to do with "concurrency". 

Sorry for noise.